File: crates/deslop/tests/fixtures/incremental-multilang/src/ledger_beta.dart
// ledger_beta.dart — the pasted copy of the Dart reconciliation routine.

class BetaLedgerCursor {
  int offset = 0;
}

int reconcileEntries(List<int> entries, int floor) {
  var balance = 0;
  for (final entry in entries) {
    if (entry > floor) {
      balance += entry * 2;
    } else {
      balance -= entry ~/ 2;
    }
  }
  return balance;
}

File: crates/deslop/tests/fixtures/incremental-multilang/src/LedgerBeta.cs
// LedgerBeta.cs — the pasted copy of the C# reconciliation routine.

namespace Ledger;

public sealed record LedgerBetaCursor(int Offset);

public sealed class LedgerBeta
{
    public static long ReconcileEntries(long[] entries, long floor)
    {
        long balance = 0;
        foreach (var entry in entries)
        {
            if (entry > floor)
            {
                balance += entry * 2;
            }
            else
            {
                balance -= entry / 2;
            }
        }
        return balance;
    }
}
